### DISCUSSION TITLE
How to get streamObject into webrtc callback in JS SDK.

### DISCUSSION THREAD

**Discussion by rahul78275** [ant-media/Ant-Media-Server#5999] [Q&A] [answered]
Hello Team

I am trying to use webrtc stream for playing into videoJs.

I want to use it like :
`this.vidojsPlayer.srcObject = stream;`

(I am aware that antmedia have videojs implementation with webrtc also).

In webRTCAdator we already pass remoteVideoId directly.

     `this.webRTCAdaptor = new WebRTCAdaptor({
      websocket_url: "wss://domain:5443/WebRTCAppEE/websocket",     
      peerconnection_config: {
          'iceServers': [{'urls': 'stun:stun1.l.google.com:19302'}]
      },=
      //remoteVideoId: "localVideo",
      isPlayMode : true,
      bandwidth: 900,
      dataChannelEnabled: true,`
      callback: (info:any, obj:any) => {
        console.log("INFO ", info, "OBJ ", obj);

help me finding the right callback to get the streamObject.

*Use case is one to many broadcast. (One streamer and multiple viewer can join)

Thank You
Rahul

**Comment by rahul78275** [comment]
@mustafaboleken @USAMAWIZARD 🔊

**Comment by USAMAWIZARD** [comment]
Please take a look this https://github.com/ant-media/videojs-webrtc-plugin 
here is a sample example
```
<script src="//path/to/video.min.js"></script>
<script src="//path/to/videojs-webrtc-plugin.min.js"></script>

<div id="video_container">
    <video id=video-player width=960 height=540 class="video-js vjs-default-skin" controls>
        <source
           src="ws://localhost:5080/LiveApp/stream1.webrtc" iceServers = '[ { "urls": "stun:stun1.l.google.com:19302" } ]'
           >
      </video>
</div>
<script>
  var player = videojs('my-video');
</script>
```

**Reply by rahul78275** [reply]
Thank you for your reply @USAMAWIZARD 
In my question i have mentioned  

> (I am aware that antmedia have videojs implementation with webrtc also).

I am looking for the **remoteVideostream** in JS SDK **callback** (like : newTrackAvailable, newStreamAvailable). With the webRTCAdaptor configuration as i mentioned in my question.

If i am unable to convey the issue, we can connect in a meet call.

**Reply by USAMAWIZARD** [reply]
Yes you can use newTrackAvailable callback is it not getting triggered ?

**Reply by rahul78275** [reply]
Yes, I do not get this callback.

**Comment by rahul78275** [comment, marked as answer]
//remoteVideoId: "localVideo",

Set remoteVideoId = null or completely removing this line did the job.

Thanks @USAMAWIZARD for identifying this.
